test/lib/e2e-utils.ts

```diff
@@ -157,6 +158,12 @@ const setupTracing = () => {
 export async function createNext(
   opts: NextInstanceOpts & { skipStart?: boolean }
 ): Promise<NextInstance> {
+  if (!semver.intersects(opts.dependencies.react, '^19.x')) {
+    opts.dependencies.react = '19.0.0-beta-4508873393-20240430'
```

Comment:
This should just throw instead. Pretty confusing if something specified is overwritten silently.

Reply:
I don't think throwing is appropriate because then it becomes the feature developers job to update fixtures (that are incorrectly versioned).

We either shouldn't be running the local Next.js instance against fixtures that aren't atleast on canary+required react version, or we do something like this that compromises.

Reply:
> I don't think throwing is appropriate because then it becomes the feature developers job to update fixtures (that are incorrectly versioned).

Why is that not appropriate? It's the feature developers responsibility to ensure the tests are correctly configured.

> shouldn't be running the local Next.js instance against fixtures that aren't atleast on canary+required react version

This seems appropriate.